### PR TITLE
Site monitoring: reduce download logs batch size and modify error notices

### PR DESCRIPTION
**Pull request by sejas**
- Closes https://github.com/Automattic/dotcom-forge/issues/3834

## Proposed Changes

* Reduce `page_size` to minimize the chances of getting an error retrieving the logs
* Change the error notices when downloading the logs

## Testing Instructions

Happy path:

* On an atomic site, navigate to `/site-monitoring/${siteSlug}/web`
* Select a wide time range to download thousands of entries.
* Click on `Download logs`
* Observe the counter increments by 500 and you download successfully.
* Click on `Error logs` and repeat the test steps.

Error messages:

* Apply this diff to your sandbox and point the public api to your machine
* Repeat the tests on the happy path
* Observe the following error messages for these error codes:
   * `500` : `There was an error retrieving logs. Please try again in a few minutes.`
   * `400`: `There was an error retrieving logs. Please try with a different time range.`

## Screenshots

**Success, using batches of five hundred entries**

https://github.com/Automattic/wp-calypso/assets/779993/264bab68-01df-4cd1-b304-6ab5cab30673

**Error 400**

Screenshot 2023-09-25 at 15 40 29

**Error 500**

Screenshot 2023-09-25 at 15 41 57

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/site-monitoring/hooks/use-site-logs-downloader.ts [hunk 1]
```diff
@@ -208,9 +208,18 @@ export const useSiteLogsDownloader = ( {
...
 					isError = true;
-					const message = get( error, 'message', 'Could not retrieve logs.' );
+					let message = get( error, 'message', 'Could not retrieve logs.' );
+					if ( error?.status === 500 ) {
+						message = translate(
+							'There was an error retrieving logs. Please try again in a few minutes.'
+						);
+					} else if ( error?.status === 400 ) {
+						message = translate(
+							'There was an error retrieving logs. Please try with a different time range.'
```
Comment (danielbachhuber) [inline client/my-sites/site-monitoring/hooks/use-site-logs-downloader.ts]:
```suggestion
							'Could not retrieve logs. Please try with a different time range.'
```

File: client/my-sites/site-monitoring/hooks/use-site-logs-downloader.ts [hunk 2]
```diff
@@ -208,9 +208,18 @@ export const useSiteLogsDownloader = ( {
...
 					} );
 				} )
-				.catch( ( error: unknown ) => {
+				.catch( ( error: { message: string; status: number } ) => {
 					isError = true;
-					const message = get( error, 'message', 'Could not retrieve logs.' );
+					let message = get( error, 'message', 'Could not retrieve logs.' );
+					if ( error?.status === 500 ) {
+						message = translate(
+							'There was an error retrieving logs. Please try again in a few minutes.'
```
Comment (danielbachhuber) [inline client/my-sites/site-monitoring/hooks/use-site-logs-downloader.ts]:
```suggestion
							'Could not retrieve logs. Please try again in a few minutes.'
```

File: client/my-sites/site-monitoring/hooks/use-site-logs-downloader.ts [hunk 3]
```diff
@@ -175,7 +175,7 @@ export const useSiteLogsDownloader = ( {
 					{
 						start: startTime,
 						end: endTime,
-						page_size: 10000,
+						page_size: 500,
```
Comment (danielbachhuber) [inline client/my-sites/site-monitoring/hooks/use-site-logs-downloader.ts:180]: Out of curiosity, how does reducing the page size resolve the error? This isn't immediately obvious to me...

**Review by danielbachhuber — COMMENTED** [review]
Seems to work as expected 👍 

I opened a support session for the original user report and easily reproduced the "Failed to retrieve Nginx source data" error:

CleanShot 2023-09-25 at 14 01 33@2x

I then loaded the support session in http://calypso.localhost:3000/ and the download process started as expected.

It looks like "Failed to retrieve Nginx source data" is the most common error.

This progress indicator is a little wonky though...

CleanShot 2023-09-25 at 13 59 18@2x

Thoughts on fixing that up? The pagination is also incorrect:

CleanShot 2023-09-25 at 14 04 30@2x

**Comment by mpkelly** [comment]
I still see an error when using a support session on a local Calpyso branch. 

I first checked prod using a support session and found there were six requests for the logs. The first five responses like this;

```json
{
    "body": {
        "code": "bad_request",
        "message": "Failed to retrieve nginx source data",
        "data": {
            "status": 500
        }
    },
    "status": 500,
    "headers": {
        "Allow": "GET"
    }
}
```

image


No error message was shown for any of these requests. The sixth request succeeded, but I could not download the logs in four attempts. 

I then tried this branch and found the table loaded instantly, and there was just one request to `sites/%a/hosting/logs`. So far, so good. But when I clicked download, I saw an error notice:

image

I clicked download again, and the downloading started, but it went way beyond the 10,000 records expected:

image

Eventually, it times out and shows a bunch of notices 😄 

image

I checked the network traffic and we are always requesting the same URL:
`https://public-api.wordpress.com/wpcom/v2/sites/%s/hosting/logs?_envelope=1&_locale=es&start=1695116356&end=1695721156&page_size=500...`

It seems like `start` and `end` are not being updated, so we keep fetching the same page. Maybe reducing the page size requires a code update elsewhere?

File: client/my-sites/site-monitoring/hooks/use-site-logs-downloader.ts (hunk 3 above)
Comment (sejas) [reply client/my-sites/site-monitoring/hooks/use-site-logs-downloader.ts:180]: My reasoning came from the intuition that a more narrow search would require fewer resources and, therefore, more chances to succeed.
Also inspired by `If they can view them in the table, why can't they download them?`

**Comment by sejas** [comment]
Thanks for reviewing this PR. It does not solve the error completely, it just increases the chances to work, and it displays better error messages.

I've hidden the total of records from the pagination and the "download progress" when we don't know the actual total (10.000+ records).

>I checked the network traffic and we are always requesting the same URL:
>https://public-api.wordpress.com/wpcom/v2/sites/%s/hosting/logs?_envelope=1&_locale=es&start=1695116356&end=1695721156&page_size=500...
>
>It seems like start and end are not being updated, so we keep fetching the same page. Maybe reducing the page size requires a code update elsewhere?

@mpkelly  The `start` and `end` do not change, but the `scroll_id` does.
I'll take a look to avoid displaying the same error multiple times 🤔 . Thanks for flagging it.

File: client/my-sites/site-monitoring/hooks/use-site-logs-downloader.ts (hunk 1 above)
Comment (sejas) [reply client/my-sites/site-monitoring/hooks/use-site-logs-downloader.ts]: done!

File: client/my-sites/site-monitoring/hooks/use-site-logs-downloader.ts (hunk 2 above)
Comment (sejas) [reply client/my-sites/site-monitoring/hooks/use-site-logs-downloader.ts]: done

**Review by danielbachhuber — APPROVED** [review]

**Comment by sejas** [comment]
In the end I decided to display the 10000 limit and actually implement the hard limit in the frontend https://github.com/Automattic/wp-calypso/pull/82134/commits/80ecbb661a834d1cfa65a35308371da5dfca7bd6

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9060366

Thank you @sejas for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9060366).

**Pull request merged**